libs/services/designer-client-services/src/lib/search.ts

```diff
@@ -40,3 +53,131 @@ export const SearchService = (): ISearchService => {
 export interface SearchResult {
   searchOperations: DiscoveryOperation<DiscoveryResultTypes>[];
 }
+
+let service2: ISearchService2;
```

Comment:
can we call it somehting other than SearchService2? Or just make these optional parts of SearchService?

Reply:
Making it part of search service would make the optional part of it a bit harder / uncleaner to split. How about `IOptionalSearchService` ?